tsinfer/inference.py

```diff
@@ -939,15 +939,34 @@ def run(self):
             logger.info("Finished building ancestors")
 
 
-def recombination_rate_to_probs(rho, positions):
-    # TODO: should max out at 0.5 or 1:  https://github.com/tskit-dev/tsinfer/issues/398
-    try:
-        return np.diff(rho.get_cumulative_mass(positions))
-    except AttributeError:
-        return np.diff(positions) * rho
-
-
 class Matcher:
+    """
+    A matching instance, used in both ``tsinfer.match_ancestors`` and
+    ``tsinfer.match_samples``. For details of the ``path_compression``,
+    `recombination_rate`` and ``mismatch_ratio`` parameters, see
+    :ref:`matching ancestors & samples<sec_inference_match_ancestors_and_samples>`.
+
+    Note that the ``recombination`` and ``mismatch`` parameters can be used in
+    ``match_ancestors`` and ``match_samples`` and are passed directly to this
+    function, but are deliberately not publicly documented in those methods.
+    They are expected to be numpy arrays of length ``num_inference_sites - 1`` and
+    ``num_inference_sites`` respectively, containing values between 0 and 1, and
+    allow recombination and mismatch probabilities to be set directly. The
+    ``recombination`` probabilities measure the probability of a recombination event
+    between adjacent inference sites, used to calculate the HMM transition probabilities
+    in the L&S-like matching algorithm. When matching a haplotype against the ancestor
+    in the immediately previous generation, ``recombination`` probabilities should reach
```

Comment:
"generation" is a bit misleading here - we don't have any concept of time and I think this will confuse people. A recombination proba of > 0.5 is always a weird thing, for example when dealing with LD calculations. This is a pretty standard issue when thinking about these extreme values for probability values in mutation and recombination processes. I think we can just say something like "Note that values > 0.5 for the recombination and mutation parameters will likely lead to pathological behaviour - for example, a mismatch probability of 1 means that a mismatch is *required* at every site.".

But nailing these parameters down as *probabilities* is very valuable and a great clarification!

Reply:
Yep. I suspect they are actually *likelihoods* (because they are conditional on a hypothesis, that this is the actual hidden state), but since everyone calls them probabilities, I guess we might as well stick with that.